WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -915,23 +916,67 @@ protected Boolean doInBackground(Void... params) {
 
         @Override
         protected void onPostExecute(Boolean saved) {
-            if (saved) {
-                Intent i = new Intent();
-                i.putExtra(EXTRA_SAVED_AS_LOCAL_DRAFT, true);
-                i.putExtra(EXTRA_IS_PAGE, mIsPage);
-                setResult(RESULT_OK, i);
-                ToastUtils.showToast(EditPostActivity.this, R.string.editor_toast_changes_saved);
-            }
+            returnResult(saved);
             finish();
         }
     }
 
-    private void saveAndFinish() {
-        if (mShowAztecEditor && mAztecEditorFragment != null) {
-            mAztecEditorFragment.saveContentFromSource();
+    private void returnResult(boolean saved) {
+        Intent i = getIntent();
+        i.putExtra(EXTRA_SAVED_AS_LOCAL_DRAFT, mPostSavedLocally);
+        i.putExtra(EXTRA_IS_PAGE, mIsPage);
+        i.putExtra(EXTRA_HAS_CHANGES, saved);
+        i.putExtra(EXTRA_IS_PUBLISHABLE, mPost.isPublishable());
+        i.putExtra(EXTRA_POSTID, mPost.getLocalTablePostId());
+        setResult(RESULT_OK, i);
+    }
+
+    private void savePostAndFinish() {
+
+        try {
+            updatePostObject(false);
+        } catch (IllegalEditorStateException e) {
+            AppLog.e(T.EDITOR, "Impossible to save and publish the post, we weren't able to update it.");
+            return;
         }
 
-        new SaveAndFinishTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        boolean shouldSave = mPost.hasChanges(mOriginalPost) && (mPost.isPublishable() || !isNewPost());
+        if (shouldSave) {
+
+            mPostSavedLocally = false;
+
+            if (isNewPost()) {
+                // new post - user just left the editor without publishing, they probably want
+                // to keep the post as a draft
+                mPost.setPostStatus(PostStatus.toString(PostStatus.DRAFT));
+                if (mEditPostSettingsFragment != null) {
+                    mEditPostSettingsFragment.updateStatusSpinner();
+                }
+            }
+
+            if (mPost.getStatusEnum() != PostStatus.PUBLISHED && mPost.isPublishable()) {
+                publishPost();
+            }
```

Comment:
Let's merge the `}` and the `else` in a single line as per [the Android style guide](https://source.android.com/source/code-style.html#use-standard-brace-style). Thanks!